File: _includes/code/java-v6/src/test/java/SearchBasicTest.java
import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.CollectionHandle;
import io.weaviate.client6.v1.api.collections.query.ConsistencyLevel;
import io.weaviate.client6.v1.api.collections.query.Metadata;
import io.weaviate.client6.v1.api.collections.query.QueryReference;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.Map;

class SearchBasicTest {

  private static WeaviateClient client;

  @BeforeAll
  public static void beforeAll() throws IOException {
    // START INSTANTIATION-COMMON
    // Best practice: store your credentials in environment variables
    String weaviateUrl = System.getenv("WEAVIATE_URL");
    String weaviateApiKey = System.getenv("WEAVIATE_API_KEY");
    String openaiApiKey = System.getenv("OPENAI_APIKEY");

    client = WeaviateClient.connectToWeaviateCloud(weaviateUrl, weaviateApiKey,
        config -> config.setHeaders(Map.of("X-OpenAI-Api-Key", openaiApiKey)));
    // END INSTANTIATION-COMMON
  }

  @AfterAll
  public static void afterAll() throws Exception {
    client.close();
  }

  @Test
  void testBasicGet() {
    // START BasicGet
    CollectionHandle<Map<String, Object>> jeopardy =
        client.collections.use("JeopardyQuestion");
    // highlight-start
    var response = jeopardy.query.fetchObjects();
    // highlight-end

    for (var o : response.objects()) {
      System.out.println(o.properties());
    }
    // END BasicGet
  }

  @Test
  void testGetWithLimit() {
    // START GetWithLimit
    CollectionHandle<Map<String, Object>> jeopardy =
        client.collections.use("JeopardyQuestion");
    var response = jeopardy.query.fetchObjects(
        // highlight-start
        q -> q.limit(1)
    // highlight-end
    );

    for (var o : response.objects()) {
      System.out.println(o.properties());
    }
    // END GetWithLimit
  }

  @Test
  void testGetWithLimitOffset() {
    // START GetWithOffset
    CollectionHandle<Map<String, Object>> jeopardy =
        client.collections.use("JeopardyQuestion");
    var response = jeopardy.query.fetchObjects(
        // highlight-start
        q -> q.limit(1).offset(1)
    // highlight-end
    );

    for (var o : response.objects()) {
      System.out.println(o.properties());
    }
    // END GetWithOffset
  }

  @Test
  void testGetProperties() {
    // START GetProperties
    CollectionHandle<Map<String, Object>> jeopardy =
        client.collections.use("JeopardyQuestion");
    var response = jeopardy.query.fetchObjects(
        // highlight-start
        q -> q.limit(1).returnProperties("question", "answer", "points")
    // highlight-end
    );

    for (var o : response.objects()) {
      System.out.println(o.properties());
    }
    // END GetProperties
  }

  @Test
  void testGetObjectVector() {
    // START GetObjectVector
    CollectionHandle<Map<String, Object>> jeopardy =
        client.collections.use("JeopardyQuestion");
    var response = jeopardy.query.fetchObjects(q -> q
        // highlight-start
        .includeVector()
        // highlight-end
        .limit(1));

    if (!response.objects().isEmpty()) {
      System.out.println(response.objects().get(0).metadata().vectors());
    }
    // END GetObjectVector
  }

  @Test
  void testGetObjectId() {
    // START GetObjectId
    CollectionHandle<Map<String, Object>> jeopardy =
        client.collections.use("JeopardyQuestion");
    var response = jeopardy.query.fetchObjects(
        // Object IDs are included by default with the v6 client! :)
        q -> q.limit(1));

    for (var o : response.objects()) {
      System.out.println(o.uuid());
    }
    // END GetObjectId
  }

  @Test
  void testGetWithCrossRefs() {
    // START GetWithCrossRefs
    CollectionHandle<Map<String, Object>> jeopardy =
        client.collections.use("JeopardyQuestion");
    var response = jeopardy.query.fetchObjects(q -> q
        // highlight-start
        .returnReferences(QueryReference.single("hasCategory",
            r -> r.returnProperties("title")))
        // highlight-end
        .limit(2));

    for (var o : response.objects()) {
      System.out.println(o.properties().get("question"));
      // print referenced objects
      if (o.references() != null && o.references().get("hasCategory") != null) {
        for (var refObj : o.references().get("hasCategory")) {
          System.out.println(refObj);
        }
      }
    }
    // END GetWithCrossRefs
  }

  @Test
  void testGetWithMetadata() {
    // START GetWithMetadata
    CollectionHandle<Map<String, Object>> jeopardy =
        client.collections.use("JeopardyQuestion");
    var response = jeopardy.query.fetchObjects(q -> q.limit(1)
        // highlight-start
        .returnMetadata(Metadata.CREATION_TIME_UNIX)
    // highlight-end
    );

    for (var o : response.objects()) {
      System.out.println(o.properties()); // View the returned properties
      System.out.println(o.metadata().creationTimeUnix()); // View the returned creation time
    }
    // END GetWithMetadata
  }

  @Test
  void testMultiTenancy() {
    // START MultiTenancy
    // Connect to the collection
    CollectionHandle<Map<String, Object>> mtCollection =
        client.collections.use("WineReviewMT");

    // Get the specific tenant's version of the collection
    // highlight-start
    var collectionTenantA = mtCollection.withTenant("tenantA");
    // highlight-end

    // Query tenantA's version
    var response = collectionTenantA.query
        .fetchObjects(q -> q.returnProperties("review_body", "title").limit(1));

    if (!response.objects().isEmpty()) {
      System.out.println(response.objects().get(0).properties());
    }
    // END MultiTenancy
  }

  @Test
  void testGetWithConsistencyLevel() {
    // START QueryWithReplication
    CollectionHandle<Map<String, Object>> jeopardy =
        client.collections.use("JeopardyQuestion")
            .withConsistencyLevel(ConsistencyLevel.QUORUM);
    // highlight-start
    var response = jeopardy.query.fetchObjectById("36ddd591-2dee-4e7e-a3cc-eb86d30a4303");
    // highlight-end

    System.out.println(response.get().properties());
    // END QueryWithReplication
  }
}

File: _includes/code/csharp/SearchBasicTest.cs
using Xunit;
using Weaviate.Client;
using Weaviate.Client.Models;
using System;
using System.Threading.Tasks;
using System.Collections.Generic;
using System.Text.Json;
using System.Linq;

// Note: This code assumes the existence of a Weaviate instance populated
// with 'JeopardyQuestion' and 'WineReviewMT' collections
public class SearchBasicTest : IAsyncLifetime
{
    private WeaviateClient client;

    public Task InitializeAsync()
    {
        // ================================
        // ===== INSTANTIATION-COMMON =====
        // ================================

        // Best practice: store your credentials in environment variables
        var weaviateUrl = Environment.GetEnvironmentVariable("WEAVIATE_URL");
        var weaviateApiKey = Environment.GetEnvironmentVariable("WEAVIATE_API_KEY");
        var openaiApiKey = Environment.GetEnvironmentVariable("OPENAI_API_KEY");

        // The Connect.Cloud helper method is a straightforward way to connect.
        // We add the OpenAI API key to the headers for the text2vec-openai module.
        client = Connect.Cloud(
            weaviateUrl,
            weaviateApiKey
        );

        return Task.CompletedTask;
    }

    public Task DisposeAsync()
    {
        // The C# client manages its connections automatically and does not require an explicit 'close' method.
        return Task.CompletedTask;
    }

    [Fact]
    public async Task BasicGet()
    {
        // ==============================
        // ===== BASIC GET EXAMPLES =====
        // ==============================

        // START BasicGet
        var jeopardy = client.Collections.Use("JeopardyQuestion");
        // highlight-start
        var response = await jeopardy.Query.FetchObjects();
        // highlight-end

        foreach (var o in response.Objects)
        {
            Console.WriteLine(JsonSerializer.Serialize(o.Properties));
        }
        // END BasicGet

        Assert.Equal("JeopardyQuestion", response.Objects.First().Collection);
        Assert.True(response.Objects.First().Properties.ContainsKey("question"));
    }

    // TODO[g-despot]: Enable when C# client supports offset
    // [Fact]
    // public async Task BasicGetOffset()
    // {
    //     // ==============================
    //     // ===== BASIC GET EXAMPLES =====
    //     // ==============================

    //     // START 
    //     var jeopardy = client.Collections.Use("JeopardyQuestion");
    //     // highlight-start
    //     var response = await jeopardy.Query.FetchObjects(offset: 1, limit: 1);
    //     // highlight-end

    //     foreach (var o in response.Objects)
    //     {
    //         Console.WriteLine(JsonSerializer.Serialize(o.Properties));
    //     }
    //     // END 

    //     Assert.Equal("JeopardyQuestion", response.Objects.First().Collection);
    //     Assert.True(response.Objects.First().Properties.ContainsKey("question"));
    // }
    // START GetWithOffset
    // Coming soon
    // END GetWithOffset

    [Fact]
    public async Task GetWithLimit()
    {
        // ====================================
        // ===== BASIC GET LIMIT EXAMPLES =====
        // ====================================

        // START GetWithLimit
        var jeopardy = client.Collections.Use<Dictionary<string, object>>("JeopardyQuestion");
        var response = await jeopardy.Query.FetchObjects(
            // highlight-start
            limit: 1
        // highlight-end
        );

        foreach (var o in response.Objects)
        {
            Console.WriteLine(JsonSerializer.Serialize(o.Properties));
        }
        // END GetWithLimit

        Assert.Equal("JeopardyQuestion", response.Objects.First().Collection);
        Assert.True(response.Objects.First().Properties.ContainsKey("question"));
        Assert.Single(response.Objects);
    }

    [Fact]
    public async Task GetProperties()
    {
        // ==========================================
        // ===== GET OBJECT PROPERTIES EXAMPLES =====
        // ==========================================

        // START GetProperties
        var jeopardy = client.Collections.Use<Dictionary<string, object>>("JeopardyQuestion");
        var response = await jeopardy.Query.FetchObjects(
            // highlight-start
            limit: 1,
            returnProperties: new[] { "question", "answer", "points" }
            // highlight-end
        );

        foreach (var o in response.Objects)
        {
            Console.WriteLine(JsonSerializer.Serialize(o.Properties));
        }
        // END GetProperties

        Assert.Equal("JeopardyQuestion", response.Objects.First().Collection);
        foreach (var propName in new[] { "question", "answer", "points" })
        {
            Assert.True(response.Objects.First().Properties.ContainsKey(propName));
        }
    }

    [Fact]
    public async Task GetObjectVector()
    {
        // ======================================
        // ===== GET OBJECT VECTOR EXAMPLES =====
        // ======================================

        // START GetObjectVector
        var jeopardy = client.Collections.Use<Dictionary<string, object>>("JeopardyQuestion");
        var response = await jeopardy.Query.FetchObjects(
            // highlight-start
            returnMetadata: (MetadataOptions.Vector, ["default"]),
            // highlight-end
            limit: 1
        );

        // Note: The C# client returns a dictionary of named vectors.
        // We assume the default vector name is 'default'.
        //TODO[g-despot]: Why is vector not returned?
        Console.WriteLine("Vector for 'default':");
        Console.WriteLine(JsonSerializer.Serialize(response.Objects.First()));
        // END GetObjectVector

        Assert.Equal("JeopardyQuestion", response.Objects.First().Collection);
        Assert.IsType<float[]>(response.Objects.First().Vectors["default"]);
    }

    [Fact]
    public async Task GetObjectId()
    {
        // ==================================
        // ===== GET OBJECT ID EXAMPLES =====
        // ==================================

        // START GetObjectId
        var jeopardy = client.Collections.Use<Dictionary<string, object>>("JeopardyQuestion");
        var response = await jeopardy.Query.FetchObjectByID(
            Guid.Parse("36ddd591-2dee-4e7e-a3cc-eb86d30a4303")
        );

        Console.WriteLine(response);
        // END GetObjectId
    }

    [Fact]
    public async Task GetWithCrossRefs()
    {
        // ==============================
        // ===== GET WITH CROSS-REF EXAMPLES =====
        // ==============================
        // START GetWithCrossRefs
        var jeopardy = client.Collections.Use<Dictionary<string, object>>("JeopardyQuestion");
        var response = await jeopardy.Query.FetchObjects(
            // highlight-start
            returnReferences: [
                new QueryReference(
                    linkOn: "hasCategory",
                    fields: ["title"]
                )
            ],
            // highlight-end
            limit: 2
        );

        foreach (var o in response.Objects)
        {
            Console.WriteLine(o.Properties["question"]);
            // print referenced objects
            // Note: References are grouped by property name ('hasCategory')
            foreach (var refObj in o.References["hasCategory"])
            {
                Console.WriteLine(JsonSerializer.Serialize(refObj.Properties));
            }
        }
        // END GetWithCrossRefs

        Assert.Equal("JeopardyQuestion", response.Objects.First().Collection);
        Assert.True(response.Objects.First().References["hasCategory"].Count > 0);
    }

    [Fact]
    public async Task GetWithMetadata()
    {
        // ====================================
        // ===== GET WITH METADATA EXAMPLE =====
        // ====================================

        // START GetWithMetadata
        var jeopardy = client.Collections.Use<Dictionary<string, object>>("JeopardyQuestion");
        var response = await jeopardy.Query.FetchObjects(
            limit: 1,
            // highlight-start
            returnMetadata: MetadataOptions.CreationTime
        // highlight-end
        );

        foreach (var o in response.Objects)
        {
            Console.WriteLine(JsonSerializer.Serialize(o.Properties));  // View the returned properties
            Console.WriteLine(o.Metadata.CreationTime);  // View the returned creation time
        }
        // END GetWithMetadata

        Assert.Equal("JeopardyQuestion", response.Objects.First().Collection);
        Assert.NotNull(response.Objects.First().Metadata.CreationTime);
    }

    [Fact]
    public async Task MultiTenancyGet()
    {
        // =========================
        // ===== MULTI-TENANCY =====
        // =========================

        // START MultiTenancy
        var mtCollection = client.Collections.Use<Dictionary<string, object>>("WineReviewMT");

        // In the C# client, the tenant is specified directly in the query method
        // rather than creating a separate tenant-specific collection object.
        // highlight-start
        var response = await mtCollection.Query.FetchObjects(
            tenant: "tenantA",
        // highlight-end
            returnProperties: new[] { "review_body", "title" },
            limit: 1
        );

        Console.WriteLine(JsonSerializer.Serialize(response.Objects.First().Properties));
        // END MultiTenancy

        Assert.True(response.Objects.Count() > 0);
        Assert.Equal("WineReviewMT", response.Objects.First().Collection);
    }

    [Fact]
    public async Task GetObjectWithReplication()
    {
        // ==================================
        // ===== GET OBJECT ID EXAMPLES =====
        // ==================================

        // START QueryWithReplication
        var jeopardy = client.Collections.Use<Dictionary<string, object>>("JeopardyQuestion").WithConsistencyLevel(ConsistencyLevels.Quorum);
        var response = await jeopardy.Query.FetchObjectByID(
            Guid.Parse("36ddd591-2dee-4e7e-a3cc-eb86d30a4303")
        );

        // The parameter passed to `withConsistencyLevel` can be one of:
        // * 'ALL',
        // * 'QUORUM' (default), or
        // * 'ONE'.
        //
        // It determines how many replicas must acknowledge a request
        // before it is considered successful.

        Console.WriteLine(response);
        // END QueryWithReplication
    }
}